src/components/InlineCodeBlock/index.tsx

```diff
@@ -1,26 +1,75 @@
 import React from 'react';
 import {StyleSheet} from 'react-native';
+import type {StyleProp, TextStyle} from 'react-native';
+import type {TDefaultRendererProps} from 'react-native-render-html';
+import EmojiWithTooltip from '@components/EmojiWithTooltip';
 import Text from '@components/Text';
-import getCurrentData from './getCurrentData';
-import type InlineCodeBlockProps from './types';
+import useThemeStyles from '@hooks/useThemeStyles';
+import type {ThemeStyles} from '@styles/index';
 import type {TTextOrTPhrasing} from './types';
+import type InlineCodeBlockProps from './types';
+
+/**
+ * This function is used to render elements based on the provided defaultRendererProps and styles.
+ * It iterates over the children of the tnode object in defaultRendererProps, and for each child,
+ * it checks if the child's tagName is 'emoji'. If it is, it creates an EmojiWithTooltip component
+ * with the appropriate styles and adds it to the elements array. If it's not, it adds the child's
+ * 'data' property to the elements array. The function then returns the elements array.
+ *
+ * @param defaultRendererProps - The default renderer props.
+ * @param textStyles - The text styles.
+ * @param styles - The theme styles.
+ * @returns The array of elements to be rendered.
+ */
+function renderElements(defaultRendererProps: TDefaultRendererProps<TTextOrTPhrasing>, textStyles: StyleProp<TextStyle>, styles: ThemeStyles) {
+    const elements: Array<string | React.JSX.Element> = [];
+
+    if ('data' in defaultRendererProps.tnode) {
+        elements.push(defaultRendererProps.tnode.data);
+        return elements;
+    }
+
+    if (!defaultRendererProps.tnode.children) {
+        return elements;
+    }
+
+    defaultRendererProps.tnode.children.forEach((child) => {
+        if (!('data' in child)) {
+            return;
+        }
+
+        if (child.tagName === 'emoji') {
+            elements.push(
+                <EmojiWithTooltip
```

Comment:
I'm not sure if we should show tooltip for emoji inside code and code block. Can you ask the design team to confirm the expected result?

Reply:
@eh2077 based on https://github.com/Expensify/App/pull/40617#issuecomment-2077007498 I think we should keep using EmojiWithTooltip (?)

Reply:
@wildan-m I'll retest it later. If we encounter tricky issue about adding tooltip, then I think we can consider remove it.